libkernelflinger/security.c

```diff
@@ -216,6 +218,123 @@ struct rot_data_t* get_rot_data()
 	return &rot_data;
 }
 
+char* strrpl(CHAR8 *in, const char *src, const char *dst){
+    CHAR8 *p = in;
+    int n = 0;
+    while(*p !='\0'){
+        if(strncmp(p, src, strlen(src)) != 0){
+            in[n] = *p;
+        }else{
+            in[n] = *dst;
+        }
+        p++;
+        n++;
+    }
+    return in;
+}
+
+CHAR8* set_attestation_ids(CHAR8* input){
+    const char *delim2 = "=";
+    CHAR8 *savedPtr2;
+    const char *d1 = ",";
+    const char *d2 = " ";
+    CHAR8 *output = strtok_r(token, delim2, (char **)&savedPtr2);
+    output = strtok_r(NULL, delim2, (char **)&savedPtr2);//Get the string after the equal sign
+    output = strrpl(output, d1, d2);//Converts comma to space in the string
+    return output;
+}
+
+/* Update the struct attestation_ids for startup_information */
+EFI_STATUS update_attestation_ids(IN VOID *vendorbootimage)
+{
+    EFI_STATUS ret = EFI_SUCCESS;
+
+    struct vendor_boot_img_hdr_v4 *vendor_hdr = (struct vendor_boot_img_hdr_v4 *)vendorbootimage;
+    UINT32 page_size = vendor_hdr->page_size;
+    UINT32 bootconfig_offset = ALIGN(sizeof(struct vendor_boot_img_hdr_v4), page_size) +
+                               ALIGN(vendor_hdr->vendor_ramdisk_size, page_size) +
+                               ALIGN(vendor_hdr->dtb_size, page_size) +
+                               ALIGN(vendor_hdr->vendor_ramdisk_table_size, page_size);
+
+    if (vendor_hdr->bootconfig_size == 0)
```

Comment:
If bootconfig_size = 0, you'd better set the default values for attestation ids, don't return a error code(EFI_UNSUPPORTED), this will halt the system.